nano/secure/ledger.cpp

```diff
@@ -400,7 +408,9 @@ void ledger_processor::epoch_block_impl (nano::state_block const & block_a)
 				}
 				if (result.code == nano::process_result::progress)
 				{
-					result.code = info.epoch () == nano::epoch::epoch_0 ? nano::process_result::progress : nano::process_result::block_position;
+					auto epoch_link = ledger.network_params.ledger.epochs.epoch (block_a.hashables.link);
+					// We either upgrade from 0 to 1, or from 1 to 2
+					result.code = (info.epoch () == nano::epoch::epoch_0 && epoch_link == nano::epoch::epoch_1) || (info.epoch () == nano::epoch::epoch_1 && epoch_link == nano::epoch::epoch_2) ? nano::process_result::progress : nano::process_result::block_position;
```

Comment:
May be `info.epoch () >= nano::epoch::epoch_0 && epoch_link == info.epoch () + 1` ?

Reply:
@cryptocode also mentioned this to me (in DM), but is it a requirement that new epochs will always follow directly after each other?

Reply:
After some discussion it was decided that epochs should be sequential so I have done this to make it easier for future epochs. Because it's an `enum class` a lot of casting is needed to compare the values so have moved the logic to a function in `epochs`, also added a test.